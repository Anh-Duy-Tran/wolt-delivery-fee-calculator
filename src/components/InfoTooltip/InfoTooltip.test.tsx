import { render, screen } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import { InfoTooltip } from '.';

describe('InfoTooltip Component', () => {
  // Snapshot test
  it('should render correct to the snapshot', () => {
    const { asFragment } = render(
      <InfoTooltip tipContent={<p>Tooltip content</p>}>Tooltip</InfoTooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should apply props class name', () => {
    render(
      <InfoTooltip
        className={'custom-class'}
        tipContent={<p>Tooltip content</p>}
      >
        Tooltip
      </InfoTooltip>
    );
    expect(screen.getByText('Tooltip')).toHaveClass('custom-class');
  });

  it('should display the expected tooltip content', () => {
    const tooltipContent = 'Tooltip content';
    render(<InfoTooltip tipContent={<p>{tooltipContent}</p>} />);
    expect(screen.getByText(tooltipContent)).toBeInTheDocument();
  });

  it('should render children correctly', () => {
    render(
      <InfoTooltip tipContent={<p>Tooltip content</p>}>
        <button>Click me</button>
      </InfoTooltip>
    );
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });

  it('should show tooltip content on hover', () => {
    render(
      <InfoTooltip
        data-testid={'info-icon'}
        tipContent={<p>Tooltip content</p>}
      >
        hover me!
      </InfoTooltip>
    );

    expect(screen.getByTestId('info-icon')).toHaveClass('peer');
    expect(screen.getByTestId('tooltip')).toHaveClass('opacity-0');
    expect(screen.getByTestId('tooltip')).toHaveClass('peer-hover:opacity-100');
  });

  it('should shows tooltip content on focus', () => {
    render(
      <InfoTooltip
        data-testid={'info-icon'}
        tipContent={<p>Tooltip content</p>}
      >
        focus me!
      </InfoTooltip>
    );

    expect(screen.getByTestId('info-icon')).toHaveClass('peer');
    expect(screen.getByTestId('tooltip')).toHaveClass('opacity-0');
    expect(screen.getByTestId('tooltip')).toHaveClass('peer-focus:opacity-100');
  });
});
