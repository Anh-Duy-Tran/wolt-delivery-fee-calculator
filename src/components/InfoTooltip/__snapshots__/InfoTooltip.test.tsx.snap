// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoTooltip Component > should render correct to the snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <div
      class="peer rounded ring-wolt-blue focus:ring-2"
      tabindex="0"
    >
      Tooltip
    </div>
    <div
      class="absolute right-0 z-50 mt-2 w-[350px] scale-75 rounded-md bg-slate-700 p-3 text-sm text-white opacity-0 transition peer-hover:scale-100 peer-hover:opacity-100 peer-focus:scale-100 peer-focus:opacity-100"
      data-testid="tooltip"
    >
      <p>
        Tooltip content
      </p>
    </div>
  </div>
</DocumentFragment>
`;
